// @keystonehq/aptossnap-adapter/build/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var utils_1 = require("./utils");
var aptos_1 = require("aptos");
var defaultSnapOrigin = 'npm:@keystonehq/aptossnap';
__exportStar(require("./utils"), exports);
var WalletAdapter = /** @class */ (function () {
    function WalletAdapter(config, snapOrigin, snapInstallationParams) {
        if (snapInstallationParams === void 0) { snapInstallationParams = {
            version: 'latest'
        }; }
        this._connecting = false;
        this.config = config;
        this.snapId = snapOrigin !== null && snapOrigin !== void 0 ? snapOrigin : defaultSnapOrigin;
        this.snapInstallationParams = snapInstallationParams;
    }
    Object.defineProperty(WalletAdapter.prototype, "connecting", {
        get: function () {
            return this._connecting;
        },
        enumerable: false,
        configurable: true
    });
    WalletAdapter.prototype.isConnected = function () {
        var _a;
        return Promise.resolve(!!((_a = this._wallet) === null || _a === void 0 ? void 0 : _a.isConnected));
    };
    WalletAdapter.prototype.account = function () {
        var _a, _b;
        if (this.connecting) {
            throw new Error('Already in connecting status');
        }
        if (!this.isConnected()) {
            throw new Error('Connect first');
        }
        else {
            return Promise.resolve({
                publicKey: ((_a = this._wallet) === null || _a === void 0 ? void 0 : _a.publicKey) || null,
                address: ((_b = this._wallet) === null || _b === void 0 ? void 0 : _b.address) || null
            });
        }
    };
    WalletAdapter.prototype.connect = function () {
        return __awaiter(this, void 0, void 0, function () {
            var isInstalled, result;
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0:
                        _b.trys.push([0, , 9, 10]);
                        if (this.connecting) {
                            throw new Error('Already in connecting status');
                        }
                        this._connecting = true;
                        // check all conditions
                        if (!utils_1.hasMetaMask()) {
                            throw new Error('Metamask is not installed');
                        }
                        return [4 /*yield*/, utils_1.isMetamaskSnapsSupported()];
                    case 1:
                        if (!(_b.sent())) {
                            throw new Error("Current Metamask version doesn't support snaps");
                        }
                        this._connecting = true;
                        return [4 /*yield*/, utils_1.isSnapInstalled(this.snapId)];
                    case 2:
                        isInstalled = _b.sent();
                        if (!!isInstalled) return [3 /*break*/, 4];
                        return [4 /*yield*/, window.ethereum.request({
                                method: 'wallet_enable',
                                params: [
                                    {
                                        wallet_snap: (_a = {},
                                            _a[this.snapId] = this.snapInstallationParams,
                                            _a)
                                    }
                                ]
                            })];
                    case 3:
                        _b.sent();
                        _b.label = 4;
                    case 4:
                        if (!!this.config.network) return [3 /*break*/, 5];
                        throw new Error('Configuration must at least define network type');
                    case 5: 
                    // sync the snap network
                    return [4 /*yield*/, window.ethereum.request({
                            method: 'wallet_invokeSnap',
                            params: [this.snapId, {
                                    method: 'aptos_configure',
                                    params: {
                                        configuration: {
                                            network: this.config.network
                                        }
                                    }
                                }]
                        })];
                    case 6:
                        // sync the snap network
                        _b.sent();
                        _b.label = 7;
                    case 7: return [4 /*yield*/, window.ethereum.request({
                            method: 'wallet_invokeSnap',
                            params: [
                                this.snapId,
                                {
                                    method: 'aptos_getAccount'
                                }
                            ]
                        })];
                    case 8:
                        result = _b.sent();
                        this._wallet = {
                            address: result.address,
                            publicKey: result.publicKey,
                            isConnected: true
                        };
                        return [2 /*return*/, result];
                    case 9:
                        this._connecting = false;
                        return [7 /*endfinally*/];
                    case 10: return [2 /*return*/];
                }
            });
        });
    };
    WalletAdapter.prototype.disconnect = function () {
        return __awaiter(this, void 0, void 0, function () {
            var wallet;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        wallet = this._wallet;
                        // clean up snap state
                        return [4 /*yield*/, window.ethereum.request({
                                method: 'wallet_invokeSnap',
                                params: [
                                    this.snapId,
                                    {
                                        method: 'aptos_disconnect'
                                    }
                                ]
                            })];
                    case 1:
                        // clean up snap state
                        _a.sent();
                        if (wallet) {
                            this._wallet = null;
                        }
                        return [2 /*return*/];
                }
            });
        });
    };
    WalletAdapter.prototype.getClient = function () {
        var nodeUrl = utils_1.getConfigurationByNetwork(this.config.network).rpc.node;
        return new aptos_1.AptosClient(nodeUrl);
    };
    WalletAdapter.prototype.signAndSubmitTransaction = function (transactionPayload, options) {
        return __awaiter(this, void 0, void 0, function () {
            var client, rawTransaction, s, signedTx;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        client = this.getClient();
                        return [4 /*yield*/, client.generateTransaction(this._wallet.address, transactionPayload, options)];
                    case 1:
                        rawTransaction = _a.sent();
                        s = new aptos_1.BCS.Serializer();
                        rawTransaction.serialize(s);
                        return [4 /*yield*/, window.ethereum.request({
                                method: 'wallet_invokeSnap',
                                params: [
                                    this.snapId,
                                    {
                                        method: 'aptos_signTransaction',
                                        params: { rawTransaction: s.getBytes() }
                                    }
                                ]
                            })];
                    case 2:
                        signedTx = _a.sent();
                        return [2 /*return*/, client.submitTransaction(new Uint8Array(Object.values(signedTx)))];
                }
            });
        });
    };
    WalletAdapter.prototype.signTransaction = function (transactionPayload) {
        return __awaiter(this, void 0, void 0, function () {
            var client, rawTransaction, s, signedTx;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        client = this.getClient();
                        return [4 /*yield*/, client.generateTransaction(this._wallet.address, transactionPayload)];
                    case 1:
                        rawTransaction = _a.sent();
                        s = new aptos_1.BCS.Serializer();
                        rawTransaction.serialize(s);
                        return [4 /*yield*/, window.ethereum.request({
                                method: 'wallet_invokeSnap',
                                params: [
                                    this.snapId,
                                    {
                                        method: 'aptos_signTransaction',
                                        params: { rawTransaction: s.getBytes() }
                                    }
                                ]
                            })];
                    case 2:
                        signedTx = _a.sent();
                        return [2 /*return*/, new Uint8Array(Object.values(signedTx))];
                }
            });
        });
    };
    WalletAdapter.prototype.signMessage = function (payload) {
        return __awaiter(this, void 0, void 0, function () {
            var client, prefix, fullMessage, address, chainId, _a, application, rawMessage, signature;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0:
                        client = this.getClient();
                        prefix = 'APTOS';
                        fullMessage = prefix;
                        address = payload.address ? this._wallet.address : undefined;
                        if (!payload.chainId) return [3 /*break*/, 2];
                        return [4 /*yield*/, client.getChainId()];
                    case 1:
                        _a = _b.sent();
                        return [3 /*break*/, 3];
                    case 2:
                        _a = undefined;
                        _b.label = 3;
                    case 3:
                        chainId = _a;
                        application = payload.application ? window.location.hostname : undefined;
                        if (address) {
                            fullMessage += "\naddress: " + address;
                        }
                        if (chainId) {
                            fullMessage += "\nchainId: " + chainId;
                        }
                        if (application) {
                            fullMessage += "\napplication: " + application;
                        }
                        fullMessage += "\nmessage: " + payload.message;
                        fullMessage += "\nnonce: " + payload.nonce;
                        rawMessage = {
                            prefix: prefix,
                            address: address,
                            chainId: chainId,
                            application: application,
                            nonce: payload.nonce,
                            message: payload.message,
                            fullMessage: fullMessage
                        };
                        Object.keys(rawMessage).forEach(function (key) { return rawMessage[key] === undefined && delete rawMessage[key]; });
                        return [4 /*yield*/, window.ethereum.request({
                                method: 'wallet_invokeSnap',
                                params: [
                                    this.snapId,
                                    {
                                        method: 'aptos_signMessage',
                                        params: { message: rawMessage }
                                    }
                                ]
                            })];
                    case 4:
                        signature = _b.sent();
                        return [2 /*return*/, __assign(__assign({}, rawMessage), { signature: "" + signature + Buffer.from(fullMessage).toString('hex') })];
                }
            });
        });
    };
    return WalletAdapter;
}());
exports.default = WalletAdapter;
