// @openblockhq/dappsdk/src/consts.js
export const OPENBLOCK_SDK_HEIGHT = 672;
export const OPENBLOCK_SDK_WIDTH = 440;
export const FIXED_SPACE_WIDTH = 12;
export const OPENBLOCK_SDK_MOBILE_HEIGHT = 600;
export const HEADER_HEIGHT = 80;
export const OPENBLOCK_SDK_ZERO_HEIGHT = 0;
export const OPENBLOCK_LOGO_LOTTIE_WIDTH = 290;
export const OPENBLOCK_LOGO_LOTTIE_HEIGHT = 80;

export const OPENBLOCK_LOGO_DARK_URL = 'https://obstatic.243096.com/download/dapp/sdk/images/logo_light.svg';
export const OPENBLOCK_LOGO_LIGHT_URL = 'https://obstatic.243096.com/download/dapp/sdk/images/logo_dark.svg';
export const ARROW_LIGHT_URL = 'https://obstatic.243096.com/download/dapp/sdk/images/arrowlight.png';
export const ARROW_DARK_URL = 'https://obstatic.243096.com/download/dapp/sdk/images/arrowblack.png';


export const StarCoinProvider = 'starcoin';
export const EVMProvider = 'evm';


export const AuthActions = [
  'eth_requestAccounts',
  'stc_requestAccounts',
  'eth_sign',
  'eth_signTypedData_v4',
  'personal_sign',
  'eth_sendTransaction',
  'stc_sendTransaction',
  'aptos_requestAccounts',
  'aptos_sign',
  'aptos_sendTransaction',
  'aptos_signTransaction',
  'sui_requestAccounts',
  'sui_signAndExecuteTransaction',
  'sui_signMessage',
  'tron_requestAccounts',
  'tron_signMessage',
  'tron_signTransaction',
  'solana_requestAccounts',
  'solana_signTransaction',
  'solana_signAndSendTransaction',
  'solana_signMessage',
  'ton_requestAccounts',
  'ton_signMessage',
  'ton_sendTransaction',
];
