// @openblockhq/dappsdk/src/data.js
const animationData = {
  v: '5.9.6',
  fr: 60,
  ip: 0,
  op: 120,
  w: 298,
  h: 80,
  nm: 'logo motion',
  ddd: 0,
  assets: [],
  layers: [
    {
      ddd: 0,
      ind: 1,
      ty: 4,
      nm: 'line',
      sr: 1,
      ks: {
        o: { a: 0, k: 100, ix: 11 },
        r: { a: 0, k: 0, ix: 10 },
        p: { a: 0, k: [149.598, 40, 0], ix: 2, l: 2 },
        a: { a: 0, k: [0, 0, 0], ix: 1, l: 2 },
        s: { a: 0, k: [100, 100, 100], ix: 6, l: 2 },
      },
      ao: 0,
      shapes: [
        {
          ind: 0,
          ty: 'sh',
          ix: 1,
          ks: {
            a: 0,
            k: {
              i: [
                [0, 0],
                [0, 0],
                [0.1, -10.2],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [10.2, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 8.5],
                [8.4, 0],
                [0, 0],
              ],
              o: [
                [0, 0],
                [10.2, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 10.1],
                [0, 0],
                [0, 0],
                [0, 0],
                [8.4, 0],
                [0, -8.4],
                [0, 0],
                [0, 0],
              ],
              v: [
                [-142.95, -25.5],
                [-96.65, -25.5],
                [-78.25, -7],
                [-78.25, -6.8],
                [-85.05, 0],
                [-78.15, 6.9],
                [-78.15, 7.1],
                [-96.65, 25.5],
                [-143.05, 25.5],
                [-129.05, 15.4],
                [-103.65, 15.6],
                [-88.15, 0],
                [-103.65, -15.6],
                [-129.05, -15.6],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 1',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 1,
          ty: 'sh',
          ix: 2,
          ks: {
            a: 0,
            k: {
              i: [
                [0, -3.2],
                [3.2, 0],
                [0, 0],
                [0, 3.2],
                [-3.2, 0],
                [0, 0],
              ],
              o: [
                [0, 3.2],
                [0, 0],
                [-3.2, 0],
                [0, -3.2],
                [0, 0],
                [3.2, 0],
              ],
              v: [
                [-97.85, 0],
                [-103.65, 5.8],
                [-125.65, 5.8],
                [-131.45, 0],
                [-125.65, -5.8],
                [-103.65, -5.8],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 2',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 2,
          ty: 'sh',
          ix: 3,
          ks: {
            a: 0,
            k: {
              i: [
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
              o: [
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
              v: [
                [68.25, -15.6],
                [72.85, -15.6],
                [72.85, 14.5],
                [68.25, 14.5],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 3',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 3,
          ty: 'sh',
          ix: 4,
          ks: {
            a: 0,
            k: {
              i: [
                [-1.8, -0.8],
                [-1.3, -1.4],
                [-0.8, -1.8],
                [0, -2.1],
                [0.8, -1.8],
                [1.4, -1.4],
                [1.8, -0.8],
                [2.1, 0],
                [1.8, 0.8],
                [1.4, 1.4],
                [0.8, 1.8],
                [0, 2.1],
                [-0.7, 1.8],
                [-1.4, 1.4],
                [-1.8, 0.7],
                [-2.1, 0],
              ],
              o: [
                [1.8, 0.7],
                [1.3, 1.4],
                [0.8, 1.8],
                [0, 2.1],
                [-0.7, 1.8],
                [-1.4, 1.3],
                [-1.8, 0.8],
                [-2.1, 0],
                [-1.8, -0.8],
                [-1.3, -1.4],
                [-0.8, -1.8],
                [0, -2.1],
                [0.8, -1.8],
                [1.4, -1.4],
                [1.9, -0.8],
                [2.2, 0],
              ],
              v: [
                [-45.65, -14.1],
                [-40.85, -10.9],
                [-37.65, -6.1],
                [-36.45, -0.2],
                [-37.65, 5.7],
                [-40.85, 10.5],
                [-45.65, 13.7],
                [-51.55, 14.9],
                [-57.55, 13.7],
                [-62.35, 10.5],
                [-65.55, 5.7],
                [-66.75, -0.2],
                [-65.65, -6.1],
                [-62.45, -10.9],
                [-57.65, -14.1],
                [-51.65, -15.3],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 4',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 4,
          ty: 'sh',
          ix: 5,
          ks: {
            a: 0,
            k: {
              i: [
                [-1.5, 0.9],
                [-0.8, 1.6],
                [-0.1, 1.9],
                [0.5, 1.3],
                [0.9, 0.9],
                [1.2, 0.5],
                [1.5, 0],
                [1.5, -0.9],
                [0.9, -1.6],
                [0, -2.1],
                [-0.5, -1.3],
                [-0.9, -0.9],
                [-1.2, -0.5],
                [-1.5, 0],
              ],
              o: [
                [1.5, -0.9],
                [0.8, -1.6],
                [0, -1.6],
                [-0.5, -1.3],
                [-0.9, -0.9],
                [-1.2, -0.5],
                [-2, 0],
                [-1.5, 0.9],
                [-0.9, 1.6],
                [0, 1.5],
                [0.5, 1.3],
                [0.9, 1],
                [1.3, 0.5],
                [2, 0],
              ],
              v: [
                [-46.25, 9.2],
                [-42.75, 5.4],
                [-41.35, 0],
                [-42.05, -4.3],
                [-44.15, -7.7],
                [-47.35, -9.9],
                [-51.45, -10.7],
                [-56.75, -9.3],
                [-60.35, -5.6],
                [-61.65, -0.1],
                [-60.95, 4.2],
                [-58.85, 7.6],
                [-55.65, 9.8],
                [-51.55, 10.6],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 5',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 5,
          ty: 'sh',
          ix: 6,
          ks: {
            a: 0,
            k: {
              i: [
                [-1.4, 0],
                [-1.4, -0.9],
                [-0.8, -1.5],
                [0, -2],
                [0.8, -1.5],
                [1.4, -0.9],
                [1.8, 0],
                [1.1, 0.6],
                [0.5, 0.5],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [-0.6, 0.3],
              ],
              o: [
                [1.8, 0],
                [1.4, 0.9],
                [0.9, 1.5],
                [0, 2],
                [-0.8, 1.6],
                [-1.4, 0.8],
                [-1.4, 0],
                [-0.7, -0.4],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0.5, -0.5],
                [1.1, -0.6],
              ],
              v: [
                [-20.45, -5.7],
                [-15.55, -4.4],
                [-12.15, -0.8],
                [-10.85, 4.5],
                [-12.15, 9.8],
                [-15.55, 13.5],
                [-20.45, 14.8],
                [-24.25, 14],
                [-26.05, 12.7],
                [-26.85, 11.9],
                [-26.85, 23.2],
                [-31.45, 23.2],
                [-31.45, -3.3],
                [-32.75, -5.2],
                [-26.75, -5.2],
                [-26.75, -2.7],
                [-25.95, -3.5],
                [-24.25, -4.8],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 6',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 6,
          ty: 'sh',
          ix: 7,
          ks: {
            a: 0,
            k: {
              i: [
                [-0.9, 0.5],
                [-0.5, 0.9],
                [-0.1, 1.2],
                [0.5, 0.9],
                [0.9, 0.5],
                [1.1, 0],
                [0.9, -0.5],
                [0.5, -0.9],
                [0, -1.2],
                [-0.5, -0.9],
                [-0.8, -0.5],
                [-1.1, 0],
              ],
              o: [
                [0.8, -0.6],
                [0.5, -1],
                [0, -1.1],
                [-0.4, -1],
                [-0.9, -0.5],
                [-1.1, 0],
                [-0.9, 0.5],
                [-0.5, 0.9],
                [0, 1.2],
                [0.5, 0.9],
                [0.9, 0.5],
                [1.1, 0],
              ],
              v: [
                [-18.35, 10],
                [-16.35, 7.8],
                [-15.55, 4.6],
                [-16.25, 1.5],
                [-18.25, -0.7],
                [-21.25, -1.5],
                [-24.25, -0.7],
                [-26.35, 1.5],
                [-27.05, 4.6],
                [-26.35, 7.8],
                [-24.35, 10],
                [-21.35, 10.8],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 7',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 7,
          ty: 'sh',
          ix: 8,
          ks: {
            a: 0,
            k: {
              i: [
                [-2, 0],
                [-1.5, -0.9],
                [-1, -1.7],
                [0, -2],
                [0.1, -0.4],
                [0.1, -0.5],
                [0, 0],
                [-0.2, -0.4],
                [-0.9, -0.5],
                [-1.2, 0],
                [-0.9, 0.3],
                [-0.6, 0.7],
                [0, 0],
                [1.5, -0.6],
                [1.7, 0],
                [1.7, 0.9],
                [1, 1.6],
                [0, 2.1],
                [-0.9, 1.6],
                [-1.7, 0.9],
              ],
              o: [
                [2, 0],
                [1.5, 0.9],
                [0.9, 1.6],
                [0, 0.4],
                [0, 0.4],
                [0, 0],
                [0.1, 0.4],
                [0.5, 0.9],
                [0.9, 0.5],
                [1.1, 0],
                [0.8, -0.3],
                [0, 0],
                [-1, 1.2],
                [-1.4, 0.6],
                [-2.2, 0],
                [-1.7, -1],
                [-1, -1.6],
                [0, -2.1],
                [1, -1.7],
                [1.6, -1],
              ],
              v: [
                [2.85, -6.3],
                [8.15, -4.9],
                [11.85, -1.1],
                [13.15, 4.2],
                [13.05, 5.4],
                [12.85, 6.7],
                [-2.35, 6.7],
                [-1.95, 7.9],
                [0.15, 10],
                [3.25, 10.7],
                [6.15, 10.2],
                [8.35, 8.6],
                [11.65, 11.9],
                [7.95, 14.6],
                [3.25, 15.5],
                [-2.55, 14.1],
                [-6.55, 10.2],
                [-8.05, 4.6],
                [-6.65, -0.9],
                [-2.65, -4.8],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 8',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 8,
          ty: 'sh',
          ix: 9,
          ks: {
            a: 0,
            k: {
              i: [
                [0.5, -0.9],
                [0.1, -0.4],
                [0, 0],
                [0.2, 0.4],
                [0.7, 0.4],
                [1.1, 0],
                [0.8, -0.5],
              ],
              o: [
                [-0.2, 0.4],
                [0, 0],
                [-0.2, -0.5],
                [-0.4, -0.8],
                [-0.8, -0.5],
                [-1.1, 0],
                [-0.8, 0.5],
              ],
              v: [
                [-2.05, 1.3],
                [-2.45, 2.4],
                [7.65, 2.3],
                [7.15, 1],
                [5.45, -0.9],
                [2.75, -1.6],
                [-0.15, -0.8],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 9',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 9,
          ty: 'sh',
          ix: 10,
          ks: {
            a: 0,
            k: {
              i: [
                [-1.5, 0],
                [-1.1, -0.7],
                [-0.4, -1.1],
                [0, -1.3],
                [0, 0],
                [0, 0],
                [0, 0],
                [0.9, 0.9],
                [1.4, 0],
                [0.7, -0.4],
                [0.4, -0.7],
                [0, -0.9],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [-0.6, 0.3],
              ],
              o: [
                [1.4, 0],
                [1.2, 0.7],
                [0.7, 1.2],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, -1.4],
                [-0.9, -0.9],
                [-0.9, 0],
                [-0.7, 0.3],
                [-0.4, 0.7],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0.4, -0.5],
                [1.2, -0.7],
              ],
              v: [
                [28.15, -5.7],
                [31.95, -4.6],
                [34.45, -1.9],
                [35.45, 1.8],
                [35.45, 14.6],
                [30.85, 14.6],
                [30.85, 3.2],
                [29.55, -0.2],
                [26.15, -1.5],
                [23.75, -0.9],
                [22.05, 0.7],
                [21.45, 3.1],
                [21.45, 14.5],
                [16.85, 14.5],
                [16.85, -3.3],
                [15.55, -5.2],
                [21.65, -5.2],
                [21.65, -2.5],
                [22.55, -3.4],
                [24.15, -4.7],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 10',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 10,
          ty: 'sh',
          ix: 11,
          ks: {
            a: 0,
            k: {
              i: [
                [0, 0],
                [0, 0],
                [-0.7, -0.8],
                [0, -1.9],
                [0.8, -1.2],
                [1.4, -0.8],
                [2.1, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [-1.3, -0.7],
                [-0.6, -1.1],
                [0, -1.3],
                [1, -1.2],
                [0.8, -0.4],
              ],
              o: [
                [0, 0],
                [1.1, 0.6],
                [1.1, 1.4],
                [0, 1.5],
                [-0.7, 1.3],
                [-1.4, 0.7],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [1.9, 0],
                [1.3, 0.7],
                [0.7, 1.1],
                [0, 1.6],
                [-0.5, 0.6],
                [0, 0],
              ],
              v: [
                [57.95, -1.2],
                [58.75, -0.8],
                [61.45, 1.3],
                [63.15, 6.2],
                [62.05, 10.3],
                [58.85, 13.3],
                [53.65, 14.4],
                [41.75, 14.4],
                [41.75, -12.9],
                [40.45, -14.8],
                [53.45, -14.8],
                [58.25, -13.7],
                [61.15, -11],
                [62.15, -7.4],
                [60.65, -3.3],
                [58.75, -1.7],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 11',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 11,
          ty: 'sh',
          ix: 12,
          ks: {
            a: 0,
            k: {
              i: [
                [1.3, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [-0.8, 0.7],
                [0, 1.4],
                [0.9, 0.7],
              ],
              o: [
                [0, 0],
                [0, 0],
                [0, 0],
                [1.3, 0],
                [0.9, -0.7],
                [0, -1.3],
                [-0.8, -0.7],
              ],
              v: [
                [52.75, -11],
                [46.55, -11],
                [46.55, -2.8],
                [52.75, -2.8],
                [55.95, -3.8],
                [57.25, -6.9],
                [55.95, -10],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 12',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 12,
          ty: 'sh',
          ix: 13,
          ks: {
            a: 0,
            k: {
              i: [
                [0, 0],
                [-0.9, 1],
                [-0.1, 1.2],
                [0.4, 0.7],
                [0.8, 0.4],
                [1, 0],
                [0, 0],
                [0, 0],
              ],
              o: [
                [1.5, 0],
                [0.9, -0.9],
                [0, -0.9],
                [-0.4, -0.8],
                [-0.7, -0.5],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
              v: [
                [53.45, 10.7],
                [57.05, 9.2],
                [58.45, 5.9],
                [57.85, 3.5],
                [56.05, 1.7],
                [53.45, 1],
                [46.55, 1],
                [46.55, 10.7],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 13',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 13,
          ty: 'sh',
          ix: 14,
          ks: {
            a: 0,
            k: {
              i: [
                [-2, 0],
                [-1.6, -0.9],
                [-0.9, -1.5],
                [0, -1.9],
                [0.9, -1.6],
                [1.6, -0.9],
                [2, 0],
                [1.6, 0.9],
                [1, 1.6],
                [0, 1.9],
                [-0.9, 1.5],
                [-1.6, 0.9],
              ],
              o: [
                [2, 0],
                [1.7, 0.9],
                [0.9, 1.5],
                [0, 1.9],
                [-0.9, 1.5],
                [-1.6, 0.9],
                [-2, 0],
                [-1.6, -0.9],
                [-0.9, -1.6],
                [0, -1.9],
                [0.9, -1.6],
                [1.6, -0.9],
              ],
              v: [
                [88.05, -5.8],
                [93.45, -4.4],
                [97.25, -0.7],
                [98.65, 4.5],
                [97.25, 9.8],
                [93.45, 13.5],
                [88.05, 14.9],
                [82.65, 13.5],
                [78.85, 9.8],
                [77.45, 4.5],
                [78.85, -0.7],
                [82.65, -4.4],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 14',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 14,
          ty: 'sh',
          ix: 15,
          ks: {
            a: 0,
            k: {
              i: [
                [-0.9, 0.5],
                [-0.5, 0.9],
                [-0.1, 1.2],
                [0.5, 0.9],
                [0.9, 0.5],
                [1.1, 0],
                [0.9, -0.5],
                [0.5, -0.9],
                [0, -1.2],
                [-0.5, -0.9],
                [-0.9, -0.5],
                [-1.1, 0],
              ],
              o: [
                [0.9, -0.6],
                [0.5, -0.9],
                [0, -1.1],
                [-0.5, -0.9],
                [-0.8, -0.5],
                [-1.1, 0],
                [-0.9, 0.5],
                [-0.5, 0.9],
                [0, 1.2],
                [0.5, 0.9],
                [0.9, 0.5],
                [1.1, 0],
              ],
              v: [
                [91.15, 9.9],
                [93.25, 7.7],
                [94.05, 4.5],
                [93.35, 1.4],
                [91.25, -0.7],
                [88.25, -1.5],
                [85.25, -0.7],
                [83.15, 1.4],
                [82.35, 4.5],
                [83.05, 7.7],
                [85.15, 9.9],
                [88.15, 10.7],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 15',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 15,
          ty: 'sh',
          ix: 16,
          ks: {
            a: 0,
            k: {
              i: [
                [-1.1, 0],
                [-0.8, 0.3],
                [-0.6, 0.6],
                [0, 0],
                [1.2, -0.5],
                [1.6, 0],
                [1.6, 0.9],
                [0.9, 1.5],
                [0, 1.9],
                [-0.9, 1.5],
                [-1.6, 0.9],
                [-2, 0],
                [-1.3, -0.6],
                [-0.9, -0.9],
                [0, 0],
                [0.7, 0.3],
                [1, 0],
                [0.9, -0.5],
                [0.5, -0.9],
                [0, -1.2],
                [-0.5, -0.9],
                [-0.9, -0.5],
              ],
              o: [
                [1, 0],
                [0.7, -0.3],
                [0, 0],
                [-1, 1],
                [-1.3, 0.6],
                [-2.1, 0],
                [-1.6, -0.9],
                [-0.9, -1.5],
                [0, -2],
                [0.9, -1.6],
                [1.7, -0.9],
                [1.5, 0],
                [1.2, 0.5],
                [0, 0],
                [-0.5, -0.4],
                [-0.8, -0.3],
                [-1.1, 0],
                [-0.9, 0.5],
                [-0.5, 0.8],
                [0, 1.2],
                [0.5, 0.9],
                [1, 0.5],
              ],
              v: [
                [112.65, 10.6],
                [115.35, 10.1],
                [117.25, 8.8],
                [120.15, 11.7],
                [116.95, 13.9],
                [112.65, 14.8],
                [107.15, 13.4],
                [103.35, 9.7],
                [101.95, 4.5],
                [103.35, -0.8],
                [107.15, -4.5],
                [112.65, -5.9],
                [116.95, -5],
                [120.05, -2.9],
                [117.15, 0],
                [115.35, -1.1],
                [112.65, -1.6],
                [109.65, -0.8],
                [107.55, 1.3],
                [106.75, 4.4],
                [107.45, 7.6],
                [109.55, 9.8],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 16',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 16,
          ty: 'sh',
          ix: 17,
          ks: {
            a: 0,
            k: {
              i: [
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
              o: [
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
              v: [
                [142.55, -5.3],
                [134.85, 4.1],
                [143.05, 14.5],
                [137.45, 14.5],
                [129.85, 4.3],
                [137.45, -5.3],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 17',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ind: 17,
          ty: 'sh',
          ix: 18,
          ks: {
            a: 0,
            k: {
              i: [
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
              o: [
                [0, 0],
                [0, 0],
                [0, 0],
                [0, 0],
              ],
              v: [
                [124.45, -15.6],
                [129.05, -15.6],
                [129.05, 14.5],
                [124.45, 14.5],
              ],
              c: true,
            },
            ix: 2,
          },
          nm: '路径 18',
          mn: 'ADBE Vector Shape - Group',
          hd: false,
        },
        {
          ty: 'st',
          c: { a: 0, k: [1, 1, 1, 1], ix: 3 },
          o: { a: 0, k: 100, ix: 4 },
          w: { a: 0, k: 1, ix: 5 },
          lc: 2,
          lj: 1,
          ml: 4,
          bm: 0,
          nm: '描边 1',
          mn: 'ADBE Vector Graphic - Stroke',
          hd: false,
        },
        {
          ty: 'tm',
          s: { a: 0, k: 0, ix: 1 },
          e: {
            a: 1,
            k: [
              {
                i: { x: [0.667], y: [1] },
                o: { x: [0.333], y: [0] },
                t: 10,
                s: [0],
              },
              { t: 60, s: [100] },
            ],
            ix: 2,
          },
          o: {
            a: 1,
            k: [
              {
                i: { x: [0.667], y: [1] },
                o: { x: [0.333], y: [0] },
                t: 10,
                s: [0],
              },
              { t: 60, s: [62] },
            ],
            ix: 3,
          },
          m: 1,
          ix: 20,
          nm: '修剪路径 1',
          mn: 'ADBE Vector Filter - Trim',
          hd: false,
        },
      ],
      ip: 0,
      op: 120,
      st: 0,
      ct: 1,
      bm: 0,
    },
    {
      ddd: 0,
      ind: 2,
      ty: 4,
      nm: 'bg',
      sr: 1,
      ks: {
        o: {
          a: 1,
          k: [
            {
              i: { x: [0.667], y: [1] },
              o: { x: [0.333], y: [0] },
              t: 50,
              s: [0],
            },
            { t: 60, s: [100] },
          ],
          ix: 11,
        },
        r: { a: 0, k: 0, ix: 10 },
        p: { a: 0, k: [149.339, 40, 0], ix: 2, l: 2 },
        a: { a: 0, k: [0, 0, 0], ix: 1, l: 2 },
        s: { a: 0, k: [100, 100, 100], ix: 6, l: 2 },
      },
      ao: 0,
      shapes: [
        {
          ty: 'gr',
          it: [
            {
              ind: 0,
              ty: 'sh',
              ix: 1,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 0],
                    [0, 0],
                    [0.101, -10.48],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [10.474, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 8.225],
                    [8.12, 0],
                    [0, 0],
                  ],
                  o: [
                    [0, 0],
                    [10.474, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 10.378],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [8.121, -0.001],
                    [0, -8.128],
                    [0, 0],
                    [0, 0],
                  ],
                  v: [
                    [-144.256, -26],
                    [-96.392, -26],
                    [-77.492, -6.997],
                    [-77.492, -6.593],
                    [-84.085, 0],
                    [-77.392, 6.693],
                    [-77.392, 7.1],
                    [-96.392, 26],
                    [-144.339, 26],
                    [-128.951, 14.899],
                    [-103.392, 15.1],
                    [-103.39, 15.1],
                    [-88.392, 0],
                    [-103.392, -15.1],
                    [-128.951, -15.1],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 1',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 1,
              ty: 'sh',
              ix: 2,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  o: [
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  v: [
                    [68.008, -16.1],
                    [73.608, -16.1],
                    [73.608, 15],
                    [68.008, 15],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 2',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 2,
              ty: 'sh',
              ix: 3,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  o: [
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  v: [
                    [124.208, -16.1],
                    [129.808, -16.1],
                    [129.808, 3.873],
                    [137.466, -5.8],
                    [143.864, -5.8],
                    [135.75, 4.106],
                    [144.339, 15],
                    [137.457, 15],
                    [129.808, 4.734],
                    [129.808, 15],
                    [124.208, 15],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 3',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 3,
              ty: 'sh',
              ix: 4,
              ks: {
                a: 0,
                k: {
                  i: [
                    [-1.87, 0.728],
                    [-2.163, 0],
                    [-1.868, -0.828],
                    [-1.354, -1.458],
                    [-0.825, -1.855],
                    [0, -2.163],
                    [0.828, -1.867],
                    [1.439, -1.439],
                    [0, 0],
                    [0, 0],
                    [1.855, -0.825],
                    [2.163, 0],
                    [1.872, 0.832],
                    [1.453, 1.453],
                    [0, 0],
                    [0, 0],
                    [0.825, 1.855],
                    [0, 2.163],
                    [-0.722, 1.856],
                    [0, 0],
                    [0, 0],
                    [-1.453, 1.453],
                  ],
                  o: [
                    [1.961, -0.825],
                    [2.257, 0],
                    [1.858, 0.724],
                    [1.334, 1.436],
                    [0.831, 1.87],
                    [0, 2.159],
                    [-0.729, 1.868],
                    [0, 0],
                    [0, 0],
                    [-1.436, 1.334],
                    [-1.87, 0.831],
                    [-2.16, 0],
                    [-1.842, -0.819],
                    [0, 0],
                    [0, 0],
                    [-1.334, -1.436],
                    [-0.831, -1.87],
                    [0, -2.163],
                    [0, 0],
                    [0, 0],
                    [0.819, -1.842],
                    [1.441, -1.441],
                  ],
                  v: [
                    [-57.579, -14.563],
                    [-51.392, -15.8],
                    [-45.199, -14.562],
                    [-40.225, -11.24],
                    [-36.935, -6.303],
                    [-35.692, -0.2],
                    [-36.93, 5.892],
                    [-40.238, 10.854],
                    [-40.245, 10.86],
                    [-40.251, 10.866],
                    [-45.189, 14.157],
                    [-51.292, 15.4],
                    [-57.495, 14.157],
                    [-62.445, 10.854],
                    [-62.452, 10.847],
                    [-62.458, 10.84],
                    [-65.749, 5.903],
                    [-66.992, -0.2],
                    [-65.858, -6.281],
                    [-65.853, -6.292],
                    [-65.849, -6.303],
                    [-62.545, -11.254],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 4',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 4,
              ty: 'sh',
              ix: 5,
              ks: {
                a: 0,
                k: {
                  i: [
                    [1.434, 0],
                    [1.417, -0.85],
                    [0.856, -1.522],
                    [0, -2.034],
                    [-0.471, -1.225],
                    [-0.837, -0.837],
                    [0, 0],
                    [0, 0],
                    [-1.144, -0.478],
                    [-1.425, 0],
                    [-1.417, 0.85],
                    [-0.758, 1.516],
                    [-0.098, 1.829],
                    [0.471, 1.223],
                    [0.837, 0.837],
                    [1.128, 0.47],
                  ],
                  o: [
                    [-1.916, 0],
                    [-1.428, 0.857],
                    [-0.849, 1.51],
                    [0, 1.46],
                    [0.483, 1.255],
                    [0, 0],
                    [0, 0],
                    [0.839, 0.932],
                    [1.252, 0.481],
                    [1.916, 0],
                    [1.417, -0.85],
                    [0.766, -1.533],
                    [-0.001, -1.553],
                    [-0.483, -1.255],
                    [-0.863, -0.863],
                    [-1.139, -0.475],
                  ],
                  v: [
                    [-51.192, -10.2],
                    [-56.234, -8.871],
                    [-59.656, -5.355],
                    [-60.892, -0.1],
                    [-60.225, 4.021],
                    [-58.238, 7.246],
                    [-58.229, 7.256],
                    [-58.22, 7.266],
                    [-55.206, 9.336],
                    [-51.292, 10.1],
                    [-46.249, 8.771],
                    [-42.939, 5.176],
                    [-41.592, -0.013],
                    [-42.258, -4.121],
                    [-44.245, -7.346],
                    [-47.284, -9.438],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 5',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 5,
              ty: 'sh',
              ix: 6,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 0],
                    [0, 0],
                    [-1.383, -0.745],
                    [-0.659, -1.199],
                    [0, -1.376],
                    [1.072, -1.286],
                    [0.878, -0.445],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [-0.745, -0.851],
                    [0, 0],
                    [0, 0],
                    [0, -2.017],
                    [0.861, -1.302],
                    [1.433, -0.819],
                    [0, 0],
                    [0, 0],
                    [2.165, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  o: [
                    [0, 0],
                    [1.971, 0],
                    [1.356, 0.73],
                    [0.751, 1.187],
                    [0, 1.722],
                    [-0.515, 0.617],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [1.143, 0.623],
                    [0, 0],
                    [0, 0],
                    [1.166, 1.484],
                    [0, 1.557],
                    [-0.763, 1.404],
                    [0, 0],
                    [0, 0],
                    [-1.48, 0.74],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  v: [
                    [39.76, -15.3],
                    [53.708, -15.3],
                    [58.745, -14.14],
                    [61.839, -11.254],
                    [62.908, -7.4],
                    [61.292, -2.98],
                    [59.254, -1.264],
                    [59.229, -1.249],
                    [59.24, -1.243],
                    [59.248, -1.239],
                    [62.085, 0.971],
                    [62.093, 0.981],
                    [62.102, 0.991],
                    [63.908, 6.2],
                    [62.737, 10.558],
                    [59.356, 13.734],
                    [59.344, 13.741],
                    [59.332, 13.747],
                    [53.908, 14.9],
                    [41.508, 14.9],
                    [41.508, -12.745],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 6',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 6,
              ty: 'sh',
              ix: 7,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [-0.677, 0.592],
                    [0, 0],
                    [0, 0],
                    [0, 1.294],
                    [0.74, 0.576],
                    [0, 0],
                    [0, 0],
                    [1.23, 0],
                  ],
                  o: [
                    [0, 0],
                    [0, 0],
                    [1.23, 0],
                    [0, 0],
                    [0, 0],
                    [0.744, -0.579],
                    [0, -1.202],
                    [0, 0],
                    [0, 0],
                    [-0.677, -0.592],
                    [0, 0],
                  ],
                  v: [
                    [47.308, -10.5],
                    [47.308, -3.3],
                    [53.008, -3.3],
                    [55.879, -4.176],
                    [55.89, -4.186],
                    [55.901, -4.195],
                    [57.008, -6.9],
                    [55.901, -9.605],
                    [55.89, -9.614],
                    [55.879, -9.624],
                    [53.008, -10.5],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 7',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 7,
              ty: 'sh',
              ix: 8,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [-0.794, 0.882],
                    [0, 0],
                    [0, 0],
                    [-0.096, 1.115],
                    [0.349, 0.611],
                    [0, 0],
                    [0, 0],
                    [0.703, 0.352],
                    [0, 0],
                    [0, 0],
                    [0.94, 0],
                  ],
                  o: [
                    [0, 0],
                    [0, 0],
                    [1.376, 0],
                    [0, 0],
                    [0, 0],
                    [0.786, -0.786],
                    [-0.003, -0.815],
                    [0, 0],
                    [0, 0],
                    [-0.352, -0.703],
                    [0, 0],
                    [0, 0],
                    [-0.585, -0.418],
                    [0, 0],
                  ],
                  v: [
                    [47.308, 1.5],
                    [47.308, 10.2],
                    [53.708, 10.2],
                    [56.937, 8.866],
                    [56.946, 8.856],
                    [56.955, 8.846],
                    [58.208, 5.879],
                    [57.674, 3.748],
                    [57.667, 3.736],
                    [57.661, 3.724],
                    [56.085, 2.147],
                    [56.05, 2.13],
                    [56.018, 2.107],
                    [53.708, 1.5],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 8',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 8,
              ty: 'sh',
              ix: 9,
              ks: {
                a: 0,
                k: {
                  i: [
                    [-1.782, 0.947],
                    [-2.077, 0],
                    [-1.583, -0.95],
                    [-1.04, -1.769],
                    [0, 0],
                    [0, -2.073],
                    [0, 0],
                    [0.101, -0.439],
                    [0.077, -0.368],
                    [0.015, -0.077],
                    [0, 0],
                    [0, 0],
                    [-0.076, -0.153],
                    [-0.815, -0.453],
                    [-1.143, 0],
                    [-0.86, 0.286],
                    [-0.553, 0.646],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [1.578, -0.632],
                    [1.755, 0],
                    [1.773, 0.939],
                    [0, 0],
                    [0, 0],
                    [1.041, 1.666],
                    [0, 2.18],
                    [-0.943, 1.676],
                    [0, 0],
                  ],
                  o: [
                    [1.685, -1.049],
                    [2.084, 0],
                    [1.575, 0.945],
                    [0, 0],
                    [0.948, 1.685],
                    [0, 0],
                    [0, 0.369],
                    [-0.008, 0.367],
                    [-0.016, 0.077],
                    [0, 0],
                    [0, 0],
                    [0.052, 0.16],
                    [0.454, 0.814],
                    [0.805, 0.447],
                    [1.029, 0],
                    [0.705, -0.266],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [-1.05, 1.259],
                    [-1.469, 0.629],
                    [-2.281, 0],
                    [0, 0],
                    [0, 0],
                    [-1.772, -1.042],
                    [-1.056, -1.69],
                    [0, -2.185],
                    [0, 0],
                    [1.039, -1.767],
                  ],
                  v: [
                    [-2.641, -5.234],
                    [3.108, -6.8],
                    [8.666, -5.329],
                    [12.539, -1.354],
                    [12.544, -1.345],
                    [13.908, 4.2],
                    [13.908, 4.223],
                    [13.808, 5.462],
                    [13.646, 6.568],
                    [13.599, 6.798],
                    [13.518, 7.2],
                    [-1.436, 7.2],
                    [-1.249, 7.667],
                    [0.651, 9.563],
                    [3.508, 10.2],
                    [6.241, 9.729],
                    [8.229, 8.275],
                    [8.58, 7.865],
                    [12.585, 11.869],
                    [12.292, 12.22],
                    [8.4, 15.062],
                    [3.508, 16],
                    [-2.526, 14.542],
                    [-2.535, 14.537],
                    [-2.545, 14.531],
                    [-6.716, 10.465],
                    [-8.292, 4.6],
                    [-6.827, -1.145],
                    [-6.823, -1.154],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 9',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 9,
              ty: 'sh',
              ix: 10,
              ks: {
                a: 0,
                k: {
                  i: [
                    [1.026, 0],
                    [0.718, -0.449],
                    [0.469, -0.841],
                    [0.045, -0.114],
                    [0, 0],
                    [0, 0],
                    [0.086, 0.171],
                    [0.595, 0.34],
                    [0, 0],
                    [0, 0],
                  ],
                  o: [
                    [-1.009, 0],
                    [-0.709, 0.443],
                    [-0.064, 0.129],
                    [0, 0],
                    [0, 0],
                    [-0.074, -0.208],
                    [-0.367, -0.734],
                    [0, 0],
                    [0, 0],
                    [-0.705, -0.441],
                  ],
                  v: [
                    [3.008, -1.1],
                    [0.373, -0.376],
                    [-1.349, 1.533],
                    [-1.511, 1.893],
                    [7.197, 1.807],
                    [7.187, 1.779],
                    [6.961, 1.224],
                    [5.46, -0.466],
                    [5.452, -0.471],
                    [5.443, -0.476],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 10',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 10,
              ty: 'sh',
              ix: 11,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 0],
                    [-2.089, 0],
                    [-1.371, -0.631],
                    [-0.94, -0.94],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0.643, 0.277],
                    [0.927, 0],
                    [0.835, -0.464],
                    [0.455, -0.818],
                    [0, 0],
                    [0, 0],
                    [0, -1.123],
                    [-0.444, -0.8],
                    [-0.794, -0.444],
                    [-0.999, 0],
                    [-0.749, 0.28],
                    [-0.553, 0.553],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [1.265, -0.528],
                    [1.663, 0],
                    [1.676, 0.943],
                    [0.943, 1.572],
                    [0, 1.987],
                    [-0.947, 1.582],
                    [-1.665, 0.937],
                  ],
                  o: [
                    [1.765, -0.934],
                    [1.561, 0],
                    [1.262, 0.527],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [-0.473, -0.378],
                    [-0.749, -0.28],
                    [-1.005, 0],
                    [-0.818, 0.455],
                    [0, 0],
                    [0, 0],
                    [-0.442, 0.707],
                    [0, 1.15],
                    [0.465, 0.837],
                    [0.947, 0.472],
                    [0.927, 0],
                    [0.636, -0.274],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [-1.037, 1.037],
                    [-1.369, 0.63],
                    [-2.185, 0],
                    [-1.678, -0.944],
                    [-0.948, -1.581],
                    [0, -2.082],
                    [0.951, -1.689],
                    [0, 0],
                  ],
                  v: [
                    [107.174, -4.942],
                    [112.908, -6.4],
                    [117.409, -5.458],
                    [120.662, -3.254],
                    [121.015, -2.9],
                    [117.445, 0.67],
                    [117.096, 0.39],
                    [115.422, -0.636],
                    [112.908, -1.1],
                    [110.151, -0.363],
                    [108.245, 1.543],
                    [108.239, 1.554],
                    [108.232, 1.565],
                    [107.508, 4.4],
                    [108.145, 7.357],
                    [110.042, 9.358],
                    [112.908, 10.1],
                    [115.422, 9.636],
                    [117.155, 8.446],
                    [117.508, 8.093],
                    [121.115, 11.7],
                    [120.762, 12.054],
                    [117.409, 14.358],
                    [112.908, 15.3],
                    [107.163, 13.836],
                    [103.18, 9.957],
                    [101.708, 4.5],
                    [103.176, -1.051],
                    [107.163, -4.936],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 11',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 11,
              ty: 'sh',
              ix: 12,
              ks: {
                a: 0,
                k: {
                  i: [
                    [0, 3.476],
                    [-3.476, 0],
                    [0, 0],
                    [0, -3.476],
                    [3.476, 0],
                    [0, 0],
                  ],
                  o: [
                    [0, -3.476],
                    [0, 0],
                    [3.476, 0],
                    [0, 3.476],
                    [0, 0],
                    [-3.476, 0],
                  ],
                  v: [
                    [-131.692, 0],
                    [-125.392, -6.3],
                    [-103.392, -6.3],
                    [-97.092, 0],
                    [-103.392, 6.3],
                    [-125.392, 6.3],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 12',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 12,
              ty: 'sh',
              ix: 13,
              ks: {
                a: 0,
                k: {
                  i: [
                    [-1.665, 0.937],
                    [-2.087, 0],
                    [-1.672, -0.939],
                    [-0.936, -1.56],
                    [0, -1.987],
                    [0.94, -1.672],
                    [0, 0],
                    [1.678, -0.944],
                    [2.087, 0],
                    [1.674, 0.941],
                    [1.04, 1.663],
                    [0, 0],
                    [0, 0],
                    [0, 1.989],
                    [-0.946, 1.579],
                  ],
                  o: [
                    [1.674, -0.941],
                    [2.085, 0],
                    [1.793, 0.951],
                    [0.948, 1.581],
                    [0, 1.989],
                    [0, 0],
                    [-0.943, 1.572],
                    [-1.674, 0.941],
                    [-2.087, 0],
                    [-1.681, -0.945],
                    [0, 0],
                    [0, 0],
                    [-0.94, -1.672],
                    [0, -1.984],
                    [0.951, -1.689],
                  ],
                  v: [
                    [82.663, -4.836],
                    [88.308, -6.3],
                    [93.948, -4.839],
                    [97.937, -0.957],
                    [99.408, 4.5],
                    [97.944, 10.045],
                    [97.937, 10.057],
                    [93.953, 13.936],
                    [88.308, 15.4],
                    [82.663, 13.936],
                    [78.684, 10.065],
                    [78.678, 10.055],
                    [78.673, 10.045],
                    [77.208, 4.5],
                    [78.676, -0.951],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 13',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 13,
              ty: 'sh',
              ix: 14,
              ks: {
                a: 0,
                k: {
                  i: [
                    [1.019, 0],
                    [0.835, -0.464],
                    [0.455, -0.818],
                    [0, -1.109],
                    [-0.444, -0.8],
                    [-0.799, -0.444],
                    [-1.005, 0],
                    [-0.83, 0.458],
                    [-0.451, 0.812],
                    [-0.099, 1.173],
                    [0.443, 0.797],
                    [0.818, 0.455],
                    [0, 0],
                    [0, 0],
                  ],
                  o: [
                    [-1.005, 0],
                    [-0.818, 0.455],
                    [-0.462, 0.831],
                    [0, 1.15],
                    [0.467, 0.841],
                    [0.835, 0.464],
                    [0.998, 0],
                    [0.839, -0.562],
                    [0.444, -0.8],
                    [-0.003, -1.039],
                    [-0.455, -0.818],
                    [0, 0],
                    [0, 0],
                    [-0.711, -0.444],
                  ],
                  v: [
                    [88.508, -1],
                    [85.751, -0.263],
                    [83.845, 1.643],
                    [83.108, 4.5],
                    [83.745, 7.457],
                    [85.651, 9.463],
                    [88.408, 10.2],
                    [91.148, 9.473],
                    [93.071, 7.457],
                    [93.808, 4.479],
                    [93.171, 1.643],
                    [91.266, -0.263],
                    [91.254, -0.269],
                    [91.243, -0.276],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 14',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 14,
              ty: 'sh',
              ix: 15,
              ks: {
                a: 0,
                k: {
                  i: [
                    [-0.643, 0.323],
                    [-1.467, 0],
                    [-1.499, -0.964],
                    [-0.846, -1.582],
                    [0, -2.06],
                    [0.836, -1.569],
                    [1.46, -0.939],
                    [0, 0],
                    [0, 0],
                    [1.892, 0],
                    [1.204, 0.656],
                    [0, 0],
                    [0.535, 0.521],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  o: [
                    [1.183, -0.643],
                    [1.871, 0],
                    [1.457, 0.936],
                    [0.954, 1.595],
                    [0, 2.082],
                    [-0.847, 1.691],
                    [0, 0],
                    [0, 0],
                    [-1.471, 0.841],
                    [-1.441, 0],
                    [0, 0],
                    [-0.692, -0.395],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0.505, -0.498],
                  ],
                  v: [
                    [-24.223, -5.243],
                    [-20.192, -6.2],
                    [-15.021, -4.821],
                    [-11.456, -1.046],
                    [-10.092, 4.5],
                    [-11.447, 10.029],
                    [-15.021, 13.921],
                    [-15.032, 13.928],
                    [-15.044, 13.934],
                    [-20.192, 15.3],
                    [-24.231, 14.439],
                    [-24.24, 14.434],
                    [-26.092, 13.106],
                    [-26.092, 23.7],
                    [-31.692, 23.7],
                    [-31.692, -3.145],
                    [-33.44, -5.7],
                    [-25.992, -5.7],
                    [-25.992, -3.907],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 15',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 15,
              ty: 'sh',
              ix: 16,
              ks: {
                a: 0,
                k: {
                  i: [
                    [1.005, 0],
                    [0.835, -0.464],
                    [0.467, -0.841],
                    [0, -1.143],
                    [-0.444, -0.8],
                    [-0.709, -0.446],
                    [-1.001, 0],
                    [-0.827, 0.454],
                    [-0.45, 0.806],
                    [-0.097, 1.152],
                    [0.443, 0.797],
                    [0, 0],
                    [0, 0],
                    [0.817, 0.454],
                  ],
                  o: [
                    [-1.005, 0],
                    [-0.799, 0.444],
                    [-0.447, 0.805],
                    [0, 1.15],
                    [0.466, 0.839],
                    [0.832, 0.46],
                    [0.993, 0],
                    [0.748, -0.565],
                    [0.454, -0.913],
                    [-0.003, -1.039],
                    [0, 0],
                    [0, 0],
                    [-0.35, -0.876],
                    [-0.835, -0.464],
                  ],
                  v: [
                    [-20.992, -1],
                    [-23.749, -0.263],
                    [-25.655, 1.743],
                    [-26.292, 4.6],
                    [-25.655, 7.557],
                    [-23.837, 9.569],
                    [-21.092, 10.3],
                    [-18.364, 9.579],
                    [-16.534, 7.567],
                    [-15.792, 4.579],
                    [-16.429, 1.743],
                    [-16.444, 1.715],
                    [-16.456, 1.686],
                    [-18.234, -0.263],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 16',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ind: 16,
              ty: 'sh',
              ix: 17,
              ks: {
                a: 0,
                k: {
                  i: [
                    [-0.654, 0.33],
                    [-1.566, 0],
                    [-1.177, -0.747],
                    [-0.46, -1.224],
                    [0, -1.371],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0.791, 0.791],
                    [1.291, 0],
                    [0.616, -0.352],
                    [0, 0],
                    [0, 0],
                    [0.359, -0.628],
                    [0, -0.823],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                  ],
                  o: [
                    [1.286, -0.747],
                    [1.493, 0],
                    [1.242, 0.726],
                    [0.73, 1.267],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, -1.291],
                    [-0.791, -0.791],
                    [-0.823, 0],
                    [0, 0],
                    [0, 0],
                    [-0.579, 0.248],
                    [-0.352, 0.616],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0, 0],
                    [0.439, -0.542],
                  ],
                  v: [
                    [24.17, -5.14],
                    [28.408, -6.2],
                    [32.469, -5.027],
                    [35.162, -2.114],
                    [36.208, 1.8],
                    [36.208, 15.1],
                    [30.608, 15.1],
                    [30.608, 3.2],
                    [29.455, 0.154],
                    [26.408, -1],
                    [24.256, -0.466],
                    [24.232, -0.452],
                    [24.205, -0.44],
                    [22.742, 0.948],
                    [22.208, 3.1],
                    [22.208, 15],
                    [16.608, 15],
                    [16.608, -3.145],
                    [14.86, -5.7],
                    [22.408, -5.7],
                    [22.408, -3.707],
                    [22.436, -3.734],
                  ],
                  c: true,
                },
                ix: 2,
              },
              nm: '路径 17',
              mn: 'ADBE Vector Shape - Group',
              hd: false,
            },
            {
              ty: 'mm',
              mm: 1,
              nm: '合并路径 1',
              mn: 'ADBE Vector Filter - Merge',
              hd: false,
            },
            {
              ty: 'fl',
              c: { a: 0, k: [1, 1, 1, 1], ix: 4 },
              o: { a: 0, k: 100, ix: 5 },
              r: 1,
              bm: 0,
              nm: '填充 1',
              mn: 'ADBE Vector Graphic - Fill',
              hd: false,
            },
            {
              ty: 'tr',
              p: { a: 0, k: [0, 0], ix: 2 },
              a: { a: 0, k: [0, 0], ix: 1 },
              s: { a: 0, k: [100, 100], ix: 3 },
              r: { a: 0, k: 0, ix: 6 },
              o: { a: 0, k: 100, ix: 7 },
              sk: { a: 0, k: 0, ix: 4 },
              sa: { a: 0, k: 0, ix: 5 },
              nm: '变换',
            },
          ],
          nm: 'bg',
          np: 19,
          cix: 2,
          bm: 0,
          ix: 1,
          mn: 'ADBE Vector Group',
          hd: false,
        },
      ],
      ip: 0,
      op: 120,
      st: 0,
      ct: 1,
      bm: 0,
    },
  ],
  markers: [],
};

export default animationData;
